hazelcast/src/main/java/com/hazelcast/internal/metrics/jmx/JmxPublisher.java

```diff
@@ -191,16 +194,16 @@ static String escapeObjectNameValue(String name) {
                     mBeanTags.append("tag").append(newTagIdx).append('=')
                              .append(escapeObjectNameValue(tag + "=" + tagValue));
                 }
-            });
+            }
             if (moduleBuilder.length() > 0) {
                 module = moduleBuilder.toString();
             }
             String discriminator = descriptor.discriminator();
             String discriminatorValue = descriptor.discriminatorValue();
             if (discriminator != null && discriminatorValue != null) {
-                mBeanTags.append(escapeObjectNameValue(discriminator))
-                         .append('=')
-                         .append(escapeObjectNameValue(discriminatorValue));
+                int newTagIdx = tagCnt.getAndInc();
+                mBeanTags.append("tag").append(newTagIdx).append("=")
```

Comment:
These changes should be probably covered by `JmxPublisherTest`. WDYT?

Reply:
@puzpuzpuz Added the coverage to the test

Reply:
Thanks!